src/Audio/CueData.cs

```diff
@@ -777,24 +962,66 @@ public void LoadTracks(AudioEngine audioEngine, List<string> waveBankNames)
 
 	internal abstract class XACTEvent
 	{
-		public uint Type
+		public uint Timestamp
 		{
 			get;
 			private set;
 		}
 
-		public uint Timestamp
+		public int Count { get; private set; }
+
+		public float Frequency { get; private set; }
+
+		public Cue Cue { get; set; }
```

Comment:
Cool, let's move this Cue instance over to a local variable in Apply() and add the aforementioned enum, then I'll go ahead and merge this.